Update readme with public info

thompiler opened the pull request:
Closes: https://github.com/airbrake/product/issues/1915

Add documentation for:
- Installation
- Authentication
- Command usage

This new readme is intended for public release and was adapted from the existing readme to the private development repo:
https://github.com/airbrake/cli

thompiler committed:
Update readme with public info

Add documentation for:
- Installation
- Authentication
- Command usage

Closes: https://github.com/airbrake/product/issues/1915

thompiler committed:
Remove extra spaces

sgray commented on README.md:
````diff
@@ -1,3 +1,113 @@
+### Homebrew
+
+```
+brew install airbrake
````
```suggestion
brew tap airbrake/airbrake-cli
brew install airbrake
```

sgray commented on README.md:
````diff
@@ -1,3 +1,113 @@
+```
+### Log in, or run commands, without a username and password
+
+If your account requires GitHub logins, or an SSO login, you cannot use the `login` command. To authenticate in this situation, you must specify either the `--user-key` or `--project-key` flags to the Airbrake CLI, or configure your `$HOME/.airbrake.yaml` file.
````
```suggestion
If your account requires GitHub or SSO login, or you have you two factor-authentication enabled, you cannot use the `login` command. To authenticate in this situation, you must specify either the `--user-key` or `--project-key` flags to the Airbrake CLI, or configure your `$HOME/.airbrake.yaml` file.
```

We should probably also mention the `airbrake config` command and its sub-commands, possibly instead of mentioning the `$HOME/.airbrake.yaml` file explicitly.

sgray commented on README.md:
````diff
@@ -1,3 +1,113 @@
+airbrake projects list
+```
+
+Another alternative is to create a config file with your user key in it. Create a file in the default location, `$HOME/.airbrake.yaml`, with the following contents:
````
Like above, I think we should point users to the `airbrake config` command and sub-commands here. `airbrake config edit` seems to make sense for what we're describing.

thompiler replied:
Agreed! Updated with mentions to the `airbrake config` command instead.

thompiler committed:
Update README.md

Co-authored-by: Steve Gray <sgray@users.noreply.github.com>

thompiler committed:
Update README.md

Co-authored-by: Steve Gray <sgray@users.noreply.github.com>

thompiler committed:
Add install command

thompiler commented:
PTAL

mmcdaris commented on README.md:
```diff
@@ -1,3 +1,125 @@
+- [Installation](#installation)
+- [Authentication](#authentication)
+  - [Username and password](#log-in-with-username-and-password)
+  - [GitHub and SSO logins](#log-in-or-run-commands-without-a-username-and-password)
```
What do you think of replacing `Github and SSO logins` with 2 sections called `Config file` and `Global  auth`? I feel like config should come before the global flags since it's more convenient and likely to be a preferred auth strategy, while global flags seem like an override that would sometimes be handy.

```diff
+ - Authentication
+   - Username and password
-   - GitHub and SSO logins
+   - Config file
+   - Global flags
```

mmcdaris commented on README.md:
````diff
@@ -1,3 +1,125 @@
+airbrake install --create-project=DESIRED_PROJECT_NAME
+```
+
+## Command usage
````
I think this probably was intended to be a parent header above the `Install Command` section.

mmcdaris commented on README.md:
```diff
@@ -1,3 +1,125 @@
+- [Authentication](#authentication)
+  - [Username and password](#log-in-with-username-and-password)
+  - [GitHub and SSO logins](#log-in-or-run-commands-without-a-username-and-password)
+- [Command usage](#command-usage)
```
What do you think of calling this section `Commands`?

mmcdaris reviewed (commented):
Nice update @thompiler, I've left some comments.

mmcdaris commented on README.md:
````diff
@@ -1,3 +1,125 @@
+```
+### Log in, or run commands, without a username and password
+
+If your account requires GitHub or SSO login, or you have you two factor-authentication enabled, you cannot use the `login` command. To authenticate in this situation, you must specify either the `--user-key` flag to the Airbrake CLI, or configure the Airbrake CLI with your user key.
````
I feel like this note should be in the Login with user/pass section above so they can avoid trying what won't work for them.

thompiler committed:
Rename headings to actual commands based on review

Do not interpret use case in the heading names or in the table 
of contents - just name the sections after the commands.

thompiler committed:
Fix note markdown

thompiler committed:
Update comments and fix links

mmcdaris reviewed (approved):
Nice addition @thompiler 👍

thompiler commented:
Thanks for the review @mmcdaris!
